tests/integration/test_multi_objective_bayesian_optimization.py

```diff
@@ -0,0 +1,77 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import gpflow
+import pytest
+import tensorflow as tf
+
+from tests.util.misc import random_seed
+from trieste.acquisition.function import ExpectedHypervolumeImprovement
+from trieste.acquisition.rule import OBJECTIVE, AcquisitionRule, EfficientGlobalOptimization
+from trieste.bayesian_optimizer import BayesianOptimizer
+from trieste.data import Dataset
+from trieste.models import GaussianProcessRegression
+from trieste.models.model_interfaces import ModelStack
+from trieste.space import Box
+from trieste.utils.multi_objectives import VLMOP2
+from trieste.utils.objectives import mk_observer
+from trieste.utils.pareto import Pareto, get_reference_point
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "num_steps, acquisition_rule",
+    [
+        (20, EfficientGlobalOptimization(ExpectedHypervolumeImprovement().using(OBJECTIVE))),
+    ],
+)
+def test_multi_objective_optimizer_finds_pareto_front_of_the_VLMOP2_function(
+    num_steps: int, acquisition_rule: AcquisitionRule
+) -> None:
+    search_space = Box([-2, -2], [2, 2])
+
+    def build_stacked_independent_objectives_model(data: Dataset) -> GaussianProcessRegression:
+        gprs = []
+        for idx in range(2):
+            single_obj_data = Dataset(
+                data.query_points, tf.gather(data.observations, [idx], axis=1)
+            )
+            variance = tf.math.reduce_variance(single_obj_data.observations)
+            kernel = gpflow.kernels.Matern52(variance, tf.constant([0.2, 0.2], tf.float64))
+            gpr = gpflow.models.GPR(single_obj_data.astuple(), kernel, noise_variance=1e-5)
+            gpflow.utilities.set_trainable(gpr.likelihood, False)
+            gprs.append((GaussianProcessRegression(gpr), 1))
+
+        return ModelStack(*gprs)
+
+    observer = mk_observer(VLMOP2().objective(), OBJECTIVE)
+
+    initial_query_points = search_space.sample(10)
+    initial_data = observer(initial_query_points)
+
+    model = build_stacked_independent_objectives_model(initial_data[OBJECTIVE])
+
+    dataset = (
+        BayesianOptimizer(observer, search_space)
+        .optimize(num_steps, initial_data, {OBJECTIVE: model}, acquisition_rule)
+        .try_get_final_datasets()[OBJECTIVE]
+    )
+
+    # A small log hypervolume difference corresponds to a succesful optimization.
+    ref_point = get_reference_point(dataset.observations)
+
+    obs_hv = Pareto(dataset.observations).hypervolume_indicator(ref_point)
+    ideal_pf = tf.cast(VLMOP2().gen_pareto_optimal_points(100), dtype=tf.float64)
+    idea_hv = Pareto(ideal_pf).hypervolume_indicator(ref_point)
+
+    assert tf.math.log(idea_hv - obs_hv) < -3.5
```

Comment:
This is an issue from original notebook (has just fixed), what we are using is $log10$ instead of $loge$ (maybe refer [here](https://botorch.org/tutorials/multi_objective_bo) for metric calculation). Then -3.5 might need also change accordingly.

Reply:
I'm using tf.math.log, so -3.5 seems to be right here?

Reply:
Ah yes numerically :)